File: VB/Styling_With_FillEmptySpace/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraPrinting

Namespace StylingWithFillEmptySpace

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private detailCount As Integer = 0

        Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            detailCount += 1
        End Sub

        Private Sub XtraReport1_FillEmptySpace(ByVal sender As Object, ByVal e As BandEventArgs)
            Dim labelsCount As Integer = Convert.ToInt32(e.Band.Height \ dummyDetailBandLabel.Height)
            Dim targetSize As Size = New Size(PageWidth - Margins.Left - Margins.Right, e.Band.Height)
            Dim labelSize As Size = New Size(dummyDetailBandLabel.Width, dummyDetailBandLabel.Height)
            Dim targetSizeInPixels As Size = GraphicsUnitConverter.Convert(targetSize, GraphicsDpi.HundredthsOfAnInch, GraphicsDpi.Pixel)
            Dim labelSizeInPixels As Size = GraphicsUnitConverter.Convert(labelSize, GraphicsDpi.HundredthsOfAnInch, GraphicsDpi.Pixel)
            Dim targetBitmap As Bitmap = New Bitmap(targetSizeInPixels.Width, targetSizeInPixels.Height)
            Dim gr As Graphics = Graphics.FromImage(targetBitmap)
            Dim pt As Point = New Point(0, 0)
            Dim sz As Size = New Size(targetSizeInPixels.Width, labelSizeInPixels.Height)
            For i As Integer = 0 To labelsCount - 1
                If i Mod 2 = 0 Then
                    gr.FillRectangle(Brushes.LightBlue, New Rectangle(pt, sz))
                Else
                    gr.FillRectangle(Brushes.Khaki, New Rectangle(pt, sz))
                End If

                pt.Y += labelSizeInPixels.Height
            Next

            Dim pictureBox As XRPictureBox = New XRPictureBox()
            pictureBox.BackColor = Color.Transparent
            pictureBox.Size = targetSize
            pictureBox.Location = New Point(0, 0)
            pictureBox.Image = targetBitmap
            e.Band.Controls.Add(pictureBox)
        End Sub
    End Class
End Namespace

File: CS/Styling_With_FillEmptySpace/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraPrinting;

namespace StylingWithFillEmptySpace {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        int detailCount = 0;

        private void Detail_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            detailCount++;
        }

        private void XtraReport1_FillEmptySpace(object sender, BandEventArgs e) {
            int labelsCount = Convert.ToInt32(e.Band.Height / this.dummyDetailBandLabel.Height);
            Size targetSize = new Size((this.PageWidth - this.Margins.Left - this.Margins.Right), e.Band.Height);
            Size labelSize = new Size(this.dummyDetailBandLabel.Width, this.dummyDetailBandLabel.Height);
            Size targetSizeInPixels = XRConvert.Convert(targetSize, GraphicsDpi.HundredthsOfAnInch, GraphicsDpi.Pixel);
            Size labelSizeInPixels = XRConvert.Convert(labelSize, GraphicsDpi.HundredthsOfAnInch, GraphicsDpi.Pixel);
            
            Bitmap targetBitmap = new Bitmap(targetSizeInPixels.Width, targetSizeInPixels.Height);
            Graphics gr = Graphics.FromImage(targetBitmap);
            Point pt = new Point(0, 0);
            Size sz = new Size(targetSizeInPixels.Width, labelSizeInPixels.Height);

            for(int i = 0; i < labelsCount; i++) {
                if(i % 2 == 0)
                    gr.FillRectangle(Brushes.LightBlue, new Rectangle(pt, sz));
                else
                    gr.FillRectangle(Brushes.Khaki, new Rectangle(pt, sz));

                pt.Y += labelSizeInPixels.Height;
            }

            XRPictureBox pictureBox = new XRPictureBox();
            pictureBox.BackColor = Color.Transparent;
            pictureBox.Size = targetSize;
            pictureBox.Location = new Point(0, 0);
            pictureBox.Image = targetBitmap;
            e.Band.Controls.Add(pictureBox);
        }
    }
}
